pandas/core/internals.py

```diff
@@ -372,6 +372,9 @@ def fillna(self, value, limit=None, inplace=False, downcast=None,
         original_value = value
         mask = isnull(self.values)
         if limit is not None:
+            if is_integer(limit):
+                assert limit > 0, ('`limit` keyword argument must be greater '
```

Comment:
raise ValueError, or is this an actual assertion? (IOW it is checked at higher levels)